src/lib/Hydra/Controller/Channel.pm

```diff
@@ -0,0 +1,83 @@
+package Hydra::Controller::Channel;
+
+use strict;
+use warnings;
+use base 'Hydra::Base::Controller::REST';
+
+
+sub channel : Chained('/') PathPart('channel/custom') CaptureArgs(3) {
+    my ($self, $c, $projectName, $jobsetName, $channelName) = @_;
+
+    $c->stash->{project} = $c->model('DB::Projects')->find($projectName);
+
+    notFound($c, "Project $projectName doesn't exist.")
+        if !$c->stash->{project};
+
+    $c->stash->{jobset} = $c->stash->{project}->jobsets->find({
+        name => $jobsetName
+    });
+
+    notFound($c, "Jobset $jobsetName doesn't exist.")
+        if !$c->stash->{jobset};
+
+    my $lastSuccessful = $c->model('DB::Builds')->find(
+        { 'eval.hasnewbuilds' => 1
+        , project => $projectName
+        , jobset => $jobsetName
+        , job => $channelName
+        , buildstatus => 0
+        },
+        { rows => 1, order_by => "eval.id desc"
+        , join => { jobsetevalmembers => 'eval' }
+        }
+    );
```

Comment:
One thing to note here (I haven't had time to look into source), with current channels we can "travel back in time" if one build is restarted, as the job is then considered not finished and it goes back and uses older finished job as latests. Has this implementation gone around that?